src/reducers/ui.js

```diff
@@ -30,6 +32,10 @@ function update(state = State(), action: Action): Record<UIState> {
     case constants.SHOW_SOURCE: {
       return state.set("shownSource", action.sourceUrl);
     }
+
+    case constants.HIGHLIGHT_SOURCE: {
```

Comment:
Haha - I think we all want to simplify this as much as possible.

We went down this path with shown source, but with highlighted source, I think it should be the same thing as selected source. Basically, every time you select a source you also want to highlight it if you can. Is that right?

If so, SourceTree can check in receiveProps if the selectedSource changed and do everything it needs to do to select it....